packages/amplify-graphql-schema-generator/src/datasource-adapter/mysql-datasource-adapter.ts

```diff
@@ -0,0 +1,146 @@
+import { Field, Index } from "../schema-representation";
+import { DataSourceAdapter } from "./datasource-adapter";
+import { knex } from 'knex';
+import { printer } from 'amplify-prompts';
+
+interface MySQLDataSourceConfig {
```

Comment:
Do we want a generic `Record<string, string>` to support other arbitrary params? e.g. keepAliveTimeouts,mimeTypes,etc?

Reply:
We will define the parameters explicitly if required to keep it strongly typed.

Reply:
We can add in the future, but a lot of RDBMS systems support MANY config variables via the url, we probably don't want users to have to wait for us to support their flags.